Documentation/quickstart/quickstart.rst

```diff
@@ -0,0 +1,110 @@
+.. include:: /substitutions.rst
+.. _quickstart:
+
+Quickstart
+==========
+
+Here's the quick version of getting started with NuttX. This is a bare-bones outline for experienced developers– if it's
+going too quickly, dive into the following sections. This Quickstart guide assumes you're on a Linux
+computer, you're using an ARM microcontroller on your embedded board, and you're familiar with using the command line.
+
+#. Install a Cross-Compiler Toolchain
+
+   With NuttX, you compile the operating system and your application on your desktop or laptop computer, then install the
+   binary file on your embedded computer. This guide assumes your computer is an
+   `ARM <https://en.wikipedia.org/wiki/ARM_architecture>`_ CPU. If it isn't, you'll need a different tool chain.
+
+   You can download a toolchain from
+   `ARM Embedded Gnu Toolchain <https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm>`_
+   for your embedded processor's CPU. You can also use a toolchain shipped with your OS for the `none-eabi` target, such as `gcc-arm-none-eabi` in Linux.
+
+
+In the following example, we download ``gcc-arm-none-eabi`` version 9.0 and unpack it into ``/opt/gcc`:
+
+   .. code-block:: console
+
+      $ sudo mkdir /opt/gcc
+      $ sudo chgrp -R users /opt/gcc
+      $ cd /opt/gcc
+      $ wget https://developer.arm.com/-/media/Files/downloads/gnu-rm/9-2019q4/gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2
+      $ tar xf gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2
+      
+Then, add the toolchain ``bin/`` directory to your path:
+
+.. code-block:: console
+
+      $ echo "export PATH=/opt/gcc/gcc-arm-none-eabi-9-2019-q4-major/bin:$PATH" >> ~/.bashrc
+
+If you are using any other shell, the procedure is similar by editing the corresponding rc file.
+
+#. Download Apache NuttX
+The next step is to download NuttX main repository along the application repository. The latter is technically optional in a very minimal configurations but should be included in normal configuration since it includes the NuttX shell.
+   .. code-block:: console
+
+      $ mkdir nuttx
+      $ cd nuttx
+      $ git clone https://github.com/apache/incubator-nuttx.git nuttx
```

Comment:
I'm thinking that maybe the quickstart should use stable release instead of cloning the repo. On the other hand, I think users are more accustomed to cloning with git. What do you think?

Reply:
@v01d Let's leave it as is, with cloning. We can add downloading a stable release later.

Reply:
@v01d @btashton The reason I think cloning is better right now is that there is no URL for the latest Apache NuttX release, for instance:

https://nuttx.apache.org/releases/latest/
https://downloads.apache.org/incubator/nuttx/latest/
https://downloads.apache.org/incubator/nuttx/latest/apache-nuttx-latest-incubating.tar.gz

(these links would always HTTP redirect to the latest release files...)

If we had that, we could put something in the docs that would not get out of date.

Reply:
There are some Apache rules about pointing people at source instead of releases, but we can come back around to that.
My guess it we will end up having the link at /docs/latest point at the latest release and /docs/dev point at master which would then point at git with a banner that says "hey these are dev docs". 

The docs that we cut for a actual release 10.0.0 should certainly point at the release tarballs.

Reply:
@btashton We can do something in sphinx and add a tag that reads from a ``VERSION`` file or something, so as long as the version is updated, the docs will be correct.

Reply:
Just a note: "latest" is used now to refer to master in the documentation. latest release would probably better named "stable".

Reply:
Yeah we can figure the naming out later, I don't think this needs to hold this up, we just need to address at the least the release docs prior to the release.

Reply:
Ok, we'll deal with this later. I'll resolve this conversation.